Realm/RLMRealm.mm

```diff
@@ -382,6 +402,17 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration er
                 }
 
                 RLMRealmSetSchemaAndAlign(realm, schema);
+            } catch (SchemaMismatchException const& exception) {
+                if (configuration.deleteRealmIfMigrationNeeded) {
+                    BOOL success = [[NSFileManager defaultManager] removeItemAtURL:configuration.fileURL error:nil];
+                    if (success) {
+                        realm = nil;
+                        return [self realmWithConfiguration:configuration error:error];
```

Comment:
I think it doesn't need to invalidate. Because schema verification error occurred before updating the schema. A transaction starts when updating the schema after schema verification was passed.

Any thought? @jpsim @tgoyne